sns/parity_math.c

```diff
@@ -96,20 +96,41 @@ static void isal_recover(struct m0_parity_math *math,
 			 enum m0_parity_linsys_algo algo);
 #endif /* ISAL_ENCODE_ENABLED */
 
+#if ISAL_ENCODE_ENABLED
+/**
+ * This is wrapper function for Intel ISA API ec_encode_data_update().
+ * @param[out] dest_buf - Array of pointers to coded output buffers
+ * @param[in]  src_buf  - Pointer to single input source used to update output
+ *                        parity.
+ * @param[in]  vec_idx  - The vector index corresponding to the single
+ *                        input source.
+ * @param[in]  g_tbls   - Pointer to array of input tables generated from
+ *                        coding coefficients in ec_init_tables().
+ *                        Must be of size 32*src_nr*dest_nr
+ * @param[in]  src_nr   - The number of vector sources for coding.
+ * @param[in]  dest_nr  - The number of output vectors to concurrently
+ *                        encode/decode.
+ * @retval     0        - success otherwise failure
+ */
+static int isal_encode_data_update(struct m0_buf *dest_buf, struct m0_buf *src_buf,
+				   uint32_t vec_idx, uint8_t *g_tbls,
+				   uint32_t src_nr, uint32_t dest_nr);
```

Comment:
Only from the definition of this function, how can we know:
1. dest_buf is pointing to an array of "struct m0_buf"?
2. src_buf is pointer to a single "struct m0_buf"?


The "src_nr" and "dest_nr" are symmetric. The  "struct m0_buf *dest_buf, struct m0_buf *src_buf" are symmetric. I suppose they point to similar memory structure.

Question:
The "src_nr" is the byte array length of this single m0_buf. Then why don't we use "src_buf->b_nob"?

Reply:
Please refer to "bufvec_to_buf_copy()"

Reply:
I will update the header as follows. I have renamed `src_nr` to `data_nr` as it was creating confusion here.
```
@param[out] dest_buf - Array of coded output buffers i.e. m0_buf
@param[in]  src_buf  - Pointer to single input source (m0_buf) used to update
                       output parity.
@param[in]  data_nr  - The number of data blocks for coding.
@param[in]  dest_nr  - The number of output blocks to concurrently
                       encode/decode.
```